File: src/main/java/org/is/bandmanager/service/PersonService.java
package org.is.bandmanager.service;

import jakarta.validation.Valid;
import org.is.bandmanager.dto.PersonDto;
import org.is.bandmanager.dto.request.PersonRequest;
import org.is.bandmanager.model.Person;

import java.util.List;

public interface PersonService {

	PersonDto create(@Valid PersonRequest request);

	List<PersonDto> getAll();

	PersonDto get(Long id);

	Person getEntity(Long id);

	PersonDto update(Long id, @Valid PersonRequest request);

	PersonDto delete(Long id);

}

File: src/app/services/person.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Person} from '../models/person.model';
import {PersonRequest} from '../models/requests/person-request.model';

@Injectable({
  providedIn: 'root'
})
export class PersonService {
  private baseUrl = '/persons';

  constructor(private http: HttpClient) {
  }

  getAll(): Observable<Person[]> {
    return this.http.get<Person[]>(this.baseUrl);
  }

  get(id: number): Observable<Person> {
    return this.http.get<Person>(`${this.baseUrl}/${id}`);
  }

  create(request: PersonRequest): Observable<Person> {
    return this.http.post<Person>(this.baseUrl, request);
  }

  update(id: number, request: PersonRequest): Observable<Person> {
    return this.http.put<Person>(`${this.baseUrl}/${id}`, request);
  }

  delete(id: number): Observable<Person> {
    return this.http.delete<Person>(`${this.baseUrl}/${id}`);
  }

}
